js/app/application.js

```diff
@@ -53,17 +66,62 @@ const Application = new Knowledge.Class({
-                             'Optional argument to set the default data path', null);
+        this.add_main_option('theme-name', 't'.charCodeAt(), GLib.OptionFlags.NONE, GLib.OptionArg.STRING,
+                             'Use a stock theme with given name instead of any application theme overrides', null);
+        this.add_main_option('default-theme', 'd'.charCodeAt(), GLib.OptionFlags.NONE, GLib.OptionArg.NONE,
+                             'Same as --theme-name=default', null);
+        this.add_main_option('recompile-theme-overrides', 'o'.charCodeAt(), GLib.OptionFlags.NONE, GLib.OptionArg.NONE,
+                             'Recompile the applications overrides.css file from the source scss', null);
+        this.add_main_option('recompile-app-json', 'j'.charCodeAt(), GLib.OptionFlags.NONE, GLib.OptionArg.NONE,
+                             'Recompile the app json file from the source yaml', null);
+        this.add_main_option('recompile-all', 'r'.charCodeAt(), GLib.OptionFlags.NONE, GLib.OptionArg.NONE,
+                             'Same as --recompile-overrides --recompile-app-json', null);
+        this.add_main_option('data-path', 'D'.charCodeAt(), GLib.OptionFlags.NONE, GLib.OptionArg.FILENAME,
+                             'Path to the default data directory for finding content', null);
+        this.add_main_option('resource-path', 'R'.charCodeAt(), GLib.OptionFlags.NONE, GLib.OptionArg.FILENAME,
+                             'Path to a different gresource to use with the application', null);
+        this.add_main_option('theme-overrides-path', 'O'.charCodeAt(), GLib.OptionFlags.NONE, GLib.OptionArg.FILENAME,
+                             'Path to a overrides scss or css file to theme the application', null);
+        this.add_main_option('app-json-path', 'J'.charCodeAt(), GLib.OptionFlags.NONE, GLib.OptionArg.FILENAME,
+                             'Path to a yaml or json file to use as a preset', null);
     },
 
     vfunc_handle_local_options: function (options) {
-        let path = options.lookup_value('data-path', null);
-        if (path)
-            Engine.get_default().default_data_path = path.deep_unpack().toString();
+        function has_option (option) {
+            return options.lookup_value(option, null) !== null;
+        }
+        function get_option_string (option) {
+            return options.lookup_value(option, null).deep_unpack().toString();
+        }
+
+        if (has_option('data-path'))
+            Engine.get_default().default_data_path = get_option_string('data-path');
+
+        if (has_option('resource-path'))
+            this.resource_path = get_option_string('resource-path');
+        let app_resource = Gio.Resource.load(this.resource_path);
+        app_resource._register();
+
+        if (has_option('default-theme'))
+            this._theme = 'default';
+        if (has_option('theme-name'))
+            this._theme = get_option_string('theme-name');
```

Comment:
If you try to use `--default-theme` and `--theme-name` at the same time, we should probably warn, instead of having the last one win.